File: src/problems/P10.py
#!/usr/bin/env python
# sum of all primes below N
from primes import is_prime

N = int(2e6)

def main(N):
	sum = 0
	
	for i in range(1, N):
		if is_prime(i):
			sum += i
			
	print("Sum of primes less than {0} is {1}".format(N, sum))
	
if __name__ == '__main__':
	main(N)

File: src/problems/P10.c
/* P10 - Sum of all primes below N. */

#include <P10.h>


int main(void)
{
   uint32_t num_primes;
   uint32_t i;
   uint32_t *primes;
   uint64_t sum_primes = 0;

   primes = prime_sieve(&num_primes, N);

   for (i = 0; i < num_primes; i++)
       sum_primes += primes[i];

   printf("Sum of primes less than %lu is %" PRIu64 "\n", N, sum_primes);

   return 0;
}

